### PR TITLE
Update and edit logsdb docs for logsdb / synthetic source GA

### PR DESCRIPTION
**Pull request by marciw**
This update focuses on licensing and core functionality for the GA release. All suggestions welcome; some will be moved to new issues (coming soon) for fast-follow or 9.0 improvements.

**Updated previews**
- [Logs data stream](https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/logs-data-stream.html) (entire topic)
- [Index settings](https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-modules.html#_static_index_settings) (index mode)
- [Source fields](https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/mapping-source-field.html) (banner only)
- [Index management](https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-mgmt.html) (screenshots)

Includes a general editing pass on changes made in https://github.com/elastic/elasticsearch/pull/115205 (which didn't get a docs team review) 😿 

Closes https://github.com/elastic/search-docs-team/issues/193

Note: I compared the docs to the in-progress blog post and didn't see any gaps, but I'll keep checking the blog post and will link to it from the docs when it's ready.

### CODE REVIEW

File: docs/reference/data-streams/logs.asciidoc [hunk 1]
```diff
@@ -1,18 +1,18 @@
 [[logs-data-stream]]
 == Logs data stream
 
-preview::[Logs data streams and the logsdb index mode are in tech preview and may be changed or removed in the future. Don't use logs data streams or logsdb index mode in production.]
+IMPORTANT: This feature requires an https://www.elastic.co/pricing[Enterprise subscription].
```
Comment (martijnvg) [inline docs/reference/data-streams/logs.asciidoc:4]: Maybe reword this? Logsdb doesn't require an Enterprise license. Without an Enterprise license stored source will be used instead of synthetic source. So an enterprise license in the case of logsdb should result into less storage usage.

File: docs/reference/index-modules.asciidoc [hunk 2]
```diff
@@ -115,8 +115,8 @@ Index mode supports the following values:
 
 `time_series`::: Index mode optimized for storage of metrics documented in <<tsds-index-settings,TSDS Settings>>.
 
-`logsdb`::: Index mode optimized for storage of logs. It applies default sort settings on the `hostname` and `timestamp` fields and uses <<synthetic-source,synthetic `_source`>>. <<index-modules-index-sorting,Index sorting>> on different fields is still allowed.
-preview:[]
+`logsdb`::: _(https://www.elastic.co/pricing[Enterprise subscription] required)_ +
+Index mode optimized for storage of logs. By default, this setting uses <<synthetic-source,synthetic `_source`>> and sorts on the `hostname` and `timestamp` fields. You can override the defaults to <<index-modules-index-sorting,sort>> on different fields.
```
Comment (leemthompo) [inline docs/reference/index-modules.asciidoc:119]:
```suggestion
<<logs-data-stream,`logsdb`>>::: Index mode optimized for storage of logs. By default, this setting uses <<synthetic-source,synthetic `_source`>> and sorts on the `hostname` and `timestamp` fields. You can override the defaults to <<index-modules-index-sorting,sort>> on different fields.
```

Comment (leemthompo) [reply docs/reference/index-modules.asciidoc:119]: Might be helpful to link to main logs db page here? I don't think you need the subscription requirement here and can delegate to the overview page for that information.

File: docs/reference/data-streams/logs.asciidoc (hunk 1 above)
Comment (leemthompo) [reply docs/reference/data-streams/logs.asciidoc:4]: I think we generally delegate to the [subscriptions page](https://www.elastic.co/subscriptions) for this info rather than being explicit in the docs. I'm guessing that page will be updated by PMM (or whoever owns that) for 8.17?

File: docs/reference/data-streams/logs.asciidoc [hunk 3]
```diff
@@ -46,154 +48,146 @@ DELETE _index_template/my-index-template
...
-reconstruct the original value in <<synthetic-source,synthetic source>>.
-
-This automatic adjustment allows synthetic source to work correctly, even when doc values are not enabled for certain
-fields.
+For example, this adjustment occurs with text fields when `store` is `false` and no suitable multi-field is available for
+reconstructing the original value.
 
 [discrete]
 [[logsdb-settings-summary]]
 === LogsDB settings summary
```
Comment (leemthompo) [inline docs/reference/data-streams/logs.asciidoc:188]:
```suggestion
=== Settings summary
```

**Review by leemthompo — COMMENTED** [review]
Changes look good to me. How we call out subscription requirements is the biggest factual question for me.

Otherwise, my main qualm with this page is the structure/headings. Here's a potential heading structure that tries to regroup info, and might make the page easier to scan in the TOC:

```markdown
Overview

Basic setup

Storage optimization
- Synthetic source
  - Fields without doc values
- Specialized codecs 

Sort settings
- Index sort defaults
- Sort orders and modes

Data validation defaults
- ignore_malformed  
- ignore_above
- ignore_dynamic_beyond_limit

Full settings reference
```

File: docs/reference/data-streams/logs.asciidoc (hunk 1 above)
Comment (leemthompo) [reply docs/reference/data-streams/logs.asciidoc:4]: Perhaps combine this note with availability information:

```asciidoc
The Elasticsearch `logsdb` index mode is generally available for Elastic Cloud Hosted 
and Self-Managed customers starting in version 8.17 and is enabled by default for 
logs in https://www.elastic.co/elasticsearch/serverless[Elastic Cloud Serverless].
```

File: docs/reference/data-streams/logs.asciidoc (hunk 3 above)
Comment (leemthompo) [reply docs/reference/data-streams/logs.asciidoc:188]: Subsequent to this, I made a [suggestion](https://github.com/elastic/elasticsearch/pull/118303#pullrequestreview-2492300187) about bigger rejiggering/renaming of sections

**Comment by marciw** [comment]
> Changes look good to me. How we call out subscription requirements is the biggest factual question for me.
> 
> Otherwise, my main qualm with this page is the structure/headings. Here's a potential heading structure that tries to regroup info, and might make the page easier to scan in the TOC:
> 
> ```
> Overview
> 
> Basic setup
> 
> Storage optimization
> - Synthetic source
>   - Fields without doc values
> - Specialized codecs 
> 
> Sort settings
> - Index sort defaults
> - Sort orders and modes
> 
> Data validation defaults
> - ignore_malformed  
> - ignore_above
> - ignore_dynamic_beyond_limit
> 
> Full settings reference
> ```

Thanks, agreed. Not sure I can tackle this now but will make it a "fast-follow" if not

**Review by shainaraskas — COMMENTED** [review]
some nits and questions for you! love the edits - it reads much clearer now

File: docs/reference/data-streams/logs.asciidoc [hunk 4]
```diff
@@ -46,154 +48,146 @@ DELETE _index_template/my-index-template
...
+create a new index. For data streams, you can update the relevant component templates and then 
+perform or wait for an index <<data-streams-rollover,rollover>>.
 
-If the default sort settings are not suitable for your use case, consider modifying them. Keep in mind that sort
-settings can influence indexing throughput, query latency, and may affect compression efficiency due to the way data
-is organized after sorting. For more details, refer to our documentation on
-<<index-modules-index-sorting,index sorting>>.
+Keep in mind that sort
+settings can influence indexing throughput and query latency, and may affect compression efficiency. For more details, see
+<<index-modules-index-sorting>>.
```
[the remark this answers is not in the document]
Comment (lkts) [reply docs/reference/data-streams/logs.asciidoc:101]: IMO this is actually the opposite. Sorting has a big impact and it is definitely beneficial to tweak sorting settings for the use case. E.g. if data does not have `host.name` field you would want to change sorting configuration.

File: docs/reference/index-modules.asciidoc [hunk 5]
```diff
@@ -113,10 +113,9 @@ Index mode supports the following values:
 
 `standard`::: Standard indexing with default settings.
 
-`time_series`::: Index mode optimized for storage of metrics documented in <<tsds-index-settings,TSDS Settings>>.
+<<tsds,`time_series`>>::: Index mode optimized for storage of metrics documented in <<tsds-index-settings,TSDS Settings>>.
 
-`logsdb`::: Index mode optimized for storage of logs. It applies default sort settings on the `hostname` and `timestamp` fields and uses <<synthetic-source,synthetic `_source`>>. <<index-modules-index-sorting,Index sorting>> on different fields is still allowed.
-preview:[]
+<<logs-data-stream,`logsdb`>>::: Index mode optimized for storage of logs. By default, this setting uses <<synthetic-source,synthetic `_source`>> and sorts on the `hostname` and `timestamp` fields. You can override the defaults to <<index-modules-index-sorting,sort>> on different fields.
```
[the remark this answers is not in the document]
Comment (lkts) [reply docs/reference/index-modules.asciidoc:118]: Agreed - i wonder if we should just say "optimized for logs" and link a page. `time_series` is like that above.

File: docs/reference/data-streams/logs.asciidoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (marciw) [reply docs/reference/data-streams/logs.asciidoc:101]: This was @salvatore-campagna 's sentence -- but I moved it and made it confusing. But I think we can lose the "warning" and just link to the index sorting topic for now (and I moved the link back to where it was, with the note about changing the sort order).

File: docs/reference/data-streams/logs.asciidoc [hunk 6]
```diff
@@ -46,154 +48,146 @@ DELETE _index_template/my-index-template
...
-named `host`, containing a `name` child field of type `keyword`. On the other hand, if `subobjects` is set to `false`,
-a single `host.name` field will be mapped as a `keyword` field.
+NOTE: If `subobjects` is set to `true` (default), the `host.name` field is mapped as an object field
+named `host` with a `name` child field of type `keyword`. If `subobjects` is set to `false`,
+a single `host.name` field is mapped as a `keyword` field.
 
-Once an index is created, the sort settings are immutable and cannot be modified. To apply different sort settings,
-a new index must be created with the desired configuration. For data streams, this can be achieved by means of an index
-rollover after updating relevant (component) templates.
+After an index is created, the sort settings cannot be modified. To use different sort settings,
```
[the remark this answers is not in the document]
Comment (marciw) [reply docs/reference/data-streams/logs.asciidoc:95]: TY -- am realizing I kind of misread the original, which just needed to be made more concise. (It was basically setting the stage for why you might need a rollover, but the extra context isn't really needed here.)

File: docs/reference/index-modules.asciidoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (marciw) [reply docs/reference/index-modules.asciidoc:118]: thank you both, agreed

@salvatore-campagna the extra sentences were yours, so just fyi

File: docs/reference/data-streams/logs.asciidoc [hunk 7]
```diff
@@ -46,154 +50,143 @@ DELETE _index_template/my-index-template
...
 [[logsdb-sort-settings]]
 === Index sort settings
 
-The following settings are applied by default when using the `logsdb` mode for index sorting:
+In `logsdb` index mode, the following sort settings are applied by default:
 
-* `index.sort.field`: `["host.name", "@timestamp"]`
-  In `logsdb` mode, indices are sorted by `host.name` and `@timestamp` fields by default. For data streams, the
-  `@timestamp` field is automatically injected if it is not present.
+`index.sort.field`: `["host.name", "@timestamp"]`::
```
Comment (salvatore-campagna) [inline docs/reference/data-streams/logs.asciidoc:74]: I was wondering if there is a good way to highlight such settings around index sorting applied by default in LogsDB and the **potential that they might cause integration issues when upgrading**, especially for custom integrations. We know that our (Elastic) integrations follow ECS and we do not have conflicts when it comes to mapping the `host.name` field but I think it is quite important to highlight that having this field automatically mapped might potentially cause issues for custom integrations (think of a customer adopting LogsDB for the first time with a new release of Elasticsearch and having a `host.name` field being a `text` field for instance).

At the moment what happens is the following: if index sort settings include `host.name` a `host.name` field will be automatically mapped as a keyword (see below for details on what is actually mapped depending on the value of `subobjects`). Because of this behaviour a customer adopting LogsDB for the first time while upgrading to the new version might experience mapping issues in case they have existing mappings using a `host.name` field mapped with a different type. This conflict leads to mapping errors, and mapping errors might result in not being able to create new indices.

IMO opinion we need to clearly state for customers transitioning to using LogsDB and adopting it for the first time, that they have at least three options here:
1. make sure, if they already have a `host.name` field, that it is mapped as a `keyword` before upgrading
2. remove `host.name` field from the set of sort fields so to avoid  `host.name` field being automatically injected by Elasticsearch (we inject the `host.name` mapping only if it is part of fields to sort on. Ideally they should choose a suitable set of fields to sort on, excluding `host.name` and maybe go for only sorting by `@timestamp` as a last-resort option).
3. opt-out of LogsDB in case there is no other way to deal with a potential mapping conflict.

I just want to restate this is IMO **critical** to highlight in the documentation since neglecting this detail might result in upgrade issues for customers moving from standard indices to LogsDB as a result of upgrading. Is there a good way to highlight this so to reduce the chances for customers to hit such a problem?

Comment (marciw) [reply docs/reference/data-streams/logs.asciidoc:74]: Thanks, will work on adding some of this

File: docs/reference/data-streams/logs.asciidoc [hunk 8]
```diff
@@ -53,16 +53,17 @@ DELETE _index_template/my-index-template
...
 
-Before using synthetic source, make sure to review the <<synthetic-source,restriction and modifications>>. To prevent modifications of a particular object or field, you can <<synthetic-source-keep,minimize synthetic source modifications>>.
+Before using synthetic source, make sure to review the <<synthetic-source,restrictions>>. 
 
 When working with multi-value fields, the `index.mapping.synthetic_source_keep` setting controls how field values
 are preserved for <<synthetic-source,synthetic source>> reconstruction. In `logsdb`, the default value is `arrays`,
 which retains both duplicate values and the order of entries. However, the exact structure of
 array elements and objects is not necessarily retained. Preserving duplicates and ordering can be critical for some log fields, such as DNS A records, HTTP headers, and log entries that represent sequential or repeated events. 
 
+If you don't have the required https://www.elastic.co/subscriptions[subscription], `logsdb` mode uses the original `_source` field.
```
Comment (shainaraskas) [inline docs/reference/data-streams/logs.asciidoc:66]: nit: I'd prefer that this sits up with the original sentence so there's no need to hunt. take or leave

File: docs/reference/data-streams/logs.asciidoc [hunk 9]
```diff
@@ -83,23 +84,19 @@ The `min` mode sorts indices by the minimum value of multi-value fields.
...
 retrieved based on the `host.name` and `@timestamp` fields.
 
-NOTE: If `subobjects` is set to `true` (default), the `host.name` field is mapped as an object field
+NOTE: If `subobjects` is set to `true` (default), the `host` field is mapped as an object field
 named `host` with a `name` child field of type `keyword`. If `subobjects` is set to `false`,
 a single `host.name` field is mapped as a `keyword` field.
 
-After an index is created, the sort settings cannot be modified. To use different sort settings,
-create a new index. For data streams, you can update the relevant component templates and then 
+To apply different sort settings, update the data stream's component templates, and then
```
Comment (shainaraskas) [inline docs/reference/data-streams/logs.asciidoc:97]: could be a followup - I see you removed the index context from this sentence. does that mean that certain index modes work with _only_ data streams? if so, we should add that info to the list of modes.

**Review by shainaraskas — APPROVED** [review]
logsdb changes lgtm! 

are the tsdb "grandfathered" nuances covered anywhere? wondering how people who have old tsdb data streams should expect new ones to behave, or should account for differences between elasticsearch instances.

File: docs/reference/data-streams/logs.asciidoc [hunk 10]
```diff
@@ -56,14 +56,12 @@ DELETE _index_template/my-index-template
 If you have the required https://www.elastic.co/subscriptions[subscription], `logsdb` index mode uses <<synthetic-source,synthetic `_source`>>, which omits storing the original `_source`
 field. Instead, the document source is synthesized from doc values or stored fields upon document retrieval. 
 
-Before using synthetic source, make sure to review the <<synthetic-source,restrictions>>. 
+If you don't have the required https://www.elastic.co/subscriptions[subscription], `logsdb` mode uses the original `_source` field.
 
-When working with multi-value fields, the `index.mapping.synthetic_source_keep` setting controls how field values
+Before using synthetic source, make sure to review the <<synthetic-source-restrictions,restrictions>>. Note also that when working with multi-value fields, the `index.mapping.synthetic_source_keep` setting controls how field values
```
Comment (shainaraskas) [inline docs/reference/data-streams/logs.asciidoc:61]: mild preference for the old two paragraph option :)

**Review by shainaraskas — APPROVED** [review]

**Comment by shainaraskas** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.17|[](https://github.com/elastic/elasticsearch/pull/118504)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Comment by shainaraskas** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/118505)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Comment by tylerperk** [comment]
Hi @marciw thanks for all the changes. I'll try to do a full review very soon. I did already notice one thing: In https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-modules.html#_static_index_settings it lists "tsds" as a valid index mode. AFAIK this is not valid and should be time_series. TSDS is the common abbreviation for that but is not an actual setting value for ES.

**Comment by marciw** [comment]
> Hi @marciw thanks for all the changes. I'll try to do a full review very soon. I did already notice one thing: In https://elasticsearch_bk_118303.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-modules.html#_static_index_settings it lists "tsds" as a valid index mode. AFAIK this is not valid and should be time_series. TSDS is the common abbreviation for that but is not an actual setting value for ES.

oof, I think that's been there for a long time -- good catch. Will fix in a separate PR.
